src/timeMachine/components/TimeMachineFluxEditor.tsx

```diff
@@ -29,10 +29,18 @@ const FluxEditor = lazy(() => import('src/shared/components/FluxMonacoEditor'))
 const TMFluxEditor: FC<{variables: Variable[]}> = props => {
   const dispatch = useDispatch()
   const activeQueryText = useSelector(getActiveQuery).text
+  const [monacoInput, setMonacoInput] = useState(activeQueryText)
+  let userIdleTimer
 
   const handleSetActiveQueryText = React.useCallback(
     (text: string) => {
-      dispatch(setActiveQueryText(text))
+      setMonacoInput(text)
+
+      // only send to redux when the user hasn't typed for one second.
+      clearTimeout(userIdleTimer)
+      userIdleTimer = setTimeout(() => {
+        dispatch(setActiveQueryText(text))
```

Comment:
This call takes 300ms to finish. Doing it every keystroke is affecting performance

Reply:
Check out the concepts of debouncing and throttling: https://blog.openreplay.com/forever-functional-debouncing-and-throttling-for-performance

and the related [`lodash.debounce`](https://lodash.com/docs/4.17.15#debounce) functionality so you don't have to roll your own